instaclustr/resource_firewall_rule.go

```diff
@@ -108,15 +113,38 @@ func resourceFirewallRuleRead(d *schema.ResourceData, meta interface{}) error {
 			d.Set("rule_cidr", value.Network)
 			d.Set("rule_security_group_id", value.SecurityGroupId)
 			d.SetId(ruleTarget)
-			d.Set("rules", value.Rules)
+
+			rules := make([]map[string]interface{}, 0)
+			for _, rule := range value.Rules {
+				ruleMapStruct := &RuleType{Type: rule.Type}
+				ruleMap, _ := StructToMap(ruleMapStruct)
+				rules = append(rules, ruleMap)
+			}
+
+			if err := d.Set("rules", rules); err != nil {
+				return fmt.Errorf("error setting rules: %s", err)
+			}
 		}
 	}
 	return nil
 }
 
+func resourceFirewallRuleStateImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	idParts := strings.Split(d.Id(), "&")
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		return nil, fmt.Errorf("Unexpected format of ID (%q), expected <CLUSTER-ID>&<RULE-CIDR>", d.Id())
+	}
+	cluster_id := idParts[0]
+	rule_cidr := idParts[1]
```

Comment:
**[Desired, Simplicity, Redundancy]** I don't see the need for additional variables here since the `d.Set()` calls explicitly identity the content of the values at the indexes of `idParts` by the field it is setting against.

Reply:
done